### DISCUSSION TITLE
Keycloak startup time, is it just me?

### DISCUSSION THREAD

**Discussion by paultid** [dasniko/testcontainers-keycloak#23] [General]
Hello,

I have started to integrate keycloak into our test framework using the your handy testcontainers implementation.  However, it is taking keycloak about 2 minutes to start up, most of that time being spent after the first message: 

`Added 'admin' to '/opt/jboss/keycloak/standalone/configuration/keycloak-add-user.json', restart server to load user -b 0.0.0.0
`

This is happening with and without the use of test container (i.e. just running from the command line).

I am using Docker on WIndows 10, and Keycloak image quay.io/keycloak/keycloak:12.0.4

Is this a known issue? Are there work arounds?

**Comment by dasniko** [comment]
Hi,

if this also occurs if you use the Keycloak container even w/o testcontainers, then I think it's not an issue with this project, but perhaps with your local Docker environment. Unfortunately I don't have any experience with Docker on Windows. I didn't see your described behaviour on Mac/Linux based systems.
You should ask on the Keycloak users mailing list or in the forum: https://www.keycloak.org/community

**Reply by paultid** [reply]
Thanks for getting back to me and confirming the startup time I am seeing is in fact unusual.  I will see if I can find some Keycloak on Docker for Windows support elsewhere.   In some ways, my problem is similar to Issue #2, except that it does start if you wait a little longer than the 2 minutes.  I worked around this by setting a 3 minute wait startup timeout on the setWaitStrategy, but obviously waiting that long is not practical.

**Reply by dasniko** [reply]
Try to set the environment variable `DB_VENDOR` explicitly to `h2`.

There is some DNS "magic" when starting the container to determine if there is a database on the network which can be used by the Keycloak container.
See https://hub.docker.com/r/jboss/keycloak/ paragraph "Database".

Maybe on your used network there is something which might cause these difficulties.

**Reply by paultid** [reply]
This worked!!!  Thanks for the idea.  Hopefully this thread will help others trying to use Keycloak in a test container.

**Reply by dasniko** [reply]
I will add this as a default, see #24.
It's not the first time, this occurs.
Thanks for starting the discussion again.
